### PR TITLE
Started logs for REST, gprc and admin

### LINKED ISSUE #186: Emit log when emulator successfully starts
### Issue To Be Solved

Currently, the flow emulator emits logs only when it transitions to "starting" state or to "error" state.

**Transition to "starting" state:**
```
> flow emulator
INFO[0000] ⚙️   Using service account 0xf8d6e0586b0a20c7  serviceAddress=f8d6e0586b0a20c7 serviceHashAlgo=SHA3_256 servicePrivKey=17db68f620ff786e3e5a8b87ba577093caece160f7ac57f5c619d86b973b7628 servicePubKey=a8d2074251eb721254004d758cf71d0bf8300651d25d6ef1991c41448ae02b8a61d232cb53a9d0f0b7adefec25fd267ed7bc09dd1490cb201f38b3b5ba783991 serviceSigAlgo=ECDSA_P256
INFO[0000] 📜  Flow contract                              FlowServiceAccount=0xf8d6e0586b0a20c7
INFO[0000] 📜  Flow contract                              FlowToken=0x0ae53cb6e3f42a79
INFO[0000] 📜  Flow contract                              FungibleToken=0xee82856bf20e2aa6
INFO[0000] 📜  Flow contract                              FlowFees=0xe5a8b7f23e8b548f
INFO[0000] 📜  Flow contract                              FlowStorageFees=0xf8d6e0586b0a20c7
INFO[0000] 🌱  Starting gRPC server on port 3569          port=3569
INFO[0000] 🌱  Starting REST API on port 8888             port=8888
INFO[0000] 🌱  Starting admin server on port 8080         port=8080
```

**Transition to "error" state:**
```
> flow emulator
INFO[0000] ⚙️   Using service account 0xf8d6e0586b0a20c7  serviceAddress=f8d6e0586b0a20c7 serviceHashAlgo=SHA3_256 servicePrivKey=17db68f620ff786e3e5a8b87ba577093caece160f7ac57f5c619d86b973b7628 servicePubKey=a8d2074251eb721254004d758cf71d0bf8300651d25d6ef1991c41448ae02b8a61d232cb53a9d0f0b7adefec25fd267ed7bc09dd1490cb201f38b3b5ba783991 serviceSigAlgo=ECDSA_P256
INFO[0000] 📜  Flow contract                              FlowFees=0xe5a8b7f23e8b548f
INFO[0000] 📜  Flow contract                              FlowStorageFees=0xf8d6e0586b0a20c7
INFO[0000] 📜  Flow contract                              FlowServiceAccount=0xf8d6e0586b0a20c7
INFO[0000] 📜  Flow contract                              FlowToken=0x0ae53cb6e3f42a79
INFO[0000] 📜  Flow contract                              FungibleToken=0xee82856bf20e2aa6
ERRO[0000] ❗  Failed to startup REST API                 error="listen tcp :8888: bind: address already in use"
```

I think it would be good to also emit a log when the emulator started successfully - when it transitions from "starting" to "started" state. This is because currently there is no indication if the emulator indeed started without errors.

### Suggest A Solution

Emit "started" info log after all the flow-emulator components were started successfully. This will indicate that no startup errors can occur from now on.

Example output:
```
INFO[0000] 🌱  Starting gRPC server on port 3569          port=3569
INFO[0000] 🌱  Starting REST API on port 8888             port=8888
INFO[0000] 🌱  Starting admin server on port 8080         port=8080

INFO[0000] ✅  Started gRPC server on port 3569          port=3569
INFO[0000] ✅  Started REST API on port 8888             port=8888
INFO[0000] ✅  Started admin server on port 8080         port=8080
```

Related code that would need to be modified: https://github.com/onflow/flow-emulator/blob/fd493635ce6984bb323ff59a32c6d8d36445e909/server/server.go#L189-L229

###  Context

The lack of this log, also makes it harder to detect successful startups programmatically. For example, at [Flowser](https://github.com/onflowser/flowser), we currently assume that if no error logs are emitted 1s after "started" logs were seen, the emulator started successfully - which is not ideal. 

See our issue: https://github.com/onflowser/flowser/issues/33

### PR DESCRIPTION
**Pull request by bluesign**
Closes #186 

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to GitHub issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [x] Re-reviewed `Files changed` in the GitHub PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: server/grpc.go [hunk 1]
```diff
@@ -76,7 +76,16 @@ func (g *GRPCServer) Start() error {
...
-	return g.grpcServer.Serve(lis)
+	g.logger.
+		WithField("port", g.port).
+		Infof("✅  Started gRPC server on port %d", g.port)
+
+	err = g.grpcServer.Serve(lis)
+	if err != nil {
+		return err
+	}
+
```
Comment (sideninja) [inline server/grpc.go:87]: should the log be after the err handling

File: server/rest.go [hunk 2]
```diff
@@ -48,13 +56,14 @@ func (r *RestServer) Stop() {
 	_ = r.server.Shutdown(context.Background())
 }
 
-func NewRestServer(be *backend.Backend, chain flow.Chain, host string, port int, debug bool) (*RestServer, error) {
-	logger := zerolog.Logger{}
+func NewRestServer(logger *logrus.Logger, be *backend.Backend, chain flow.Chain, host string, port int, debug bool) (*RestServer, error) {
+
+	debug_logger := zerolog.Logger{}
```
Comment (sideninja) [inline server/rest.go:61]:
```suggestion
	debugLogger := zerolog.Logger{}
```

**Comment by sideninja** [comment]
@bluesign left couple of comments

File: server/grpc.go (hunk 1 above)
Comment (bluesign) [reply server/grpc.go:87]: Serve is a blocking call. Basically if we opened the port for listening, kernel should queue incoming connections and requests, So if we can listen the port, we assume started.

Comment (sideninja) [reply server/grpc.go:87]: oh right sorry, forgot.

File: server/rest.go [hunk 3]
```diff
@@ -48,13 +56,14 @@ func (r *RestServer) Stop() {
...
 }
 
-func NewRestServer(be *backend.Backend, chain flow.Chain, host string, port int, debug bool) (*RestServer, error) {
-	logger := zerolog.Logger{}
+func NewRestServer(logger *logrus.Logger, be *backend.Backend, chain flow.Chain, host string, port int, debug bool) (*RestServer, error) {
+
+	debugLogger := zerolog.Logger{}
 	if debug {
-		logger = zerolog.New(os.Stdout)
+		debug_logger = zerolog.New(os.Stdout)
```
Comment (sideninja) [inline server/rest.go:63]: rename those as well

**Review by sideninja — APPROVED** [review]
Awesome, thank you

**Pull request merged**